src/modules/Experiences/ExperiencesList/helpers.js

```diff
@@ -84,4 +84,61 @@ const configureForm = (expType) => {
   }
 };
 
-export { formatMonthYearDate, getWorkLength, configureForm };
+const monthScore = {
+  January: 1,
+  February: 2,
+  March: 3,
+  April: 4,
+  May: 5,
+  June: 6,
+  July: 7,
+  August: 8,
+  September: 9,
+  October: 10,
+  November: 11,
+  December: 12,
+};
+
+const getMonthScore = (experiences) => {
+  experiences.map((exp) => {
+    exp.start_month_score = monthScore[exp.start_month];
+
+    // exclude end_month === 'none'
+    if (exp.is_current === false) {
+      exp.end_month_score = monthScore[exp.end_month];
+    }
+  });
+
+  return experiences;
+};
+
+const sortExperiences = (experiences) => {
+  const experiencesWithScores = getMonthScore(experiences);
+
+  const sortedExperiences = experiencesWithScores.sort(function(exp1, exp2) {
+    // if is_current is True or have the same end month/year
+    // sort from start month/year (newest experience)
+    if (exp1.end_month === exp2.end_month && exp1.end_year === exp2.end_year) {
+      if (exp1.start_year > exp2.start_year) return -1;
+      if (exp1.start_year < exp2.start_year) return 1;
+      if (exp1.start_month_score > exp2.start_month_score) return -1;
+      if (exp1.start_month_score < exp2.start_month_score) return 1;
+    } else if (exp1.is_current === false && exp2.is_current === false) {
+      if (exp1.end_year > exp2.end_year) return -1;
+      if (exp1.end_year < exp2.end_year) return 1;
+      if (exp1.end_month_score > exp2.end_month_score) return -1;
+      if (exp1.end_month_score < exp2.end_month_score) return 1;
+    }
+    if (exp1.is_current === true) return -1;
+  });
```

Comment:
Also, one obvious bug here is that you can hit the end of the function without returning a number. You should always ensure your functions end with an unconditional `return` (or an `if/else` with a return in both branches)